client/post-editor/editor-featured-image/dropzone.jsx

```diff
@@ -0,0 +1,114 @@
+import { head, uniqueId } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import DropZone from 'components/drop-zone';
+import MediaActions from 'lib/media/actions';
+import MediaStore from 'lib/media/store';
+import MediaUtils from 'lib/media/utils';
+import PostActions from 'lib/posts/actions';
+
+import { receiveMedia, deleteMedia } from 'state/media/actions';
+import { editPost } from 'state/posts/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+
+class FeaturedImageDropZone extends Component {
+	static propTypes = {
+		site: PropTypes.object.isRequired,
+		post: PropTypes.object.isRequired,
+	};
+
+	static contextTypes = {
+		store: PropTypes.object
+	};
+
+	onFilesDrop = ( files ) => {
+		/**
+		 * Filter files for `image` media prefix and return the first image.
+		 *
+		 * At the moment we ignore all the other images that were dragged onto the DropZone
+		 */
+		const droppedImage = head( MediaUtils.filterItemsByMimePrefix( files, 'image' ) );
+
+		if ( ! droppedImage ) {
+			return false;
+		}
+
+		const transientMediaId = uniqueId( 'featured-image' );
+		const siteID = this.props.siteId;
+
+		const handleFeaturedImageUpload = () => {
+			const media = MediaStore.get( siteID, transientMediaId );
+			const isUploadInProgress = media && MediaUtils.isItemBeingUploaded( media );
+			const isFailedUpload = ! media;
+
+			if ( isFailedUpload ) {
+				this.props.deleteMedia( siteID, transientMediaId );
+			} else {
+				this.props.receiveMedia( siteID, media );
+			}
+
+			/**
+			 * File uploaded successfully, there is no longer need to listen for the changes.
+			 */
+			if ( media && ! isUploadInProgress ) {
+				MediaStore.off( 'change', handleFeaturedImageUpload );
+			}
+
+			/**
+			 * TODO: Redux way. What's left: figure out how to properly use `editPost`
+			 * and research if the whole FeaturedImage component has to be updated to
+			 * work properly with Redux.
+			 *
+			 * Right now `PostActions.edit` seems to be the best way to approach the problem.
+			 */
+			// this.props.editPost( siteID, this.props.postId, { featured_image: media.ID } );
+
+			// Cannot dispatch an action while in a dispatched action. Temporary(tm).
+			setTimeout( () => {
+				PostActions.edit( {
+					featured_image: media.ID
+				} );
+			}, 0 );
+		};
+
+		MediaStore.on( 'change', handleFeaturedImageUpload );
+
+		MediaActions.add( this.props.site.ID, {
+			ID: transientMediaId,
+			fileContents: droppedImage,
+			fileName: droppedImage.name
+		} );
+	};
+
+	render() {
+		return (
+			<DropZone
+				onFilesDrop={ this.onFilesDrop }
+				textLabel={ this.props.translate( 'Drop an image to set as a Featured Image' ) }
+			/>
+		);
+	}
+}
+
+export default connect(
+	( state ) => {
+		const siteId = getSelectedSiteId( state );
+		const postId = getEditorPostId( state );
+
+		return { siteId, postId };
+	},
```

Comment:
Thanks for catching that. It's a leftover code, because at the start, the code was structured around the legacy way of accessing `site`/`post`. Throughout the iterations, the access to these properties have been updated to use Redux and I forgot to remove the unnecessary references to them.
The code has been updated to properly only use the Redux state selectors.